crates/ty_python_semantic/src/types/liskov.rs

```diff
@@ -43,43 +45,24 @@ fn check_class_declaration<'db>(
 
     let MemberWithDefinition { member, definition } = member;
 
-    // TODO: Check Liskov on non-methods too
-    let Type::FunctionLiteral(function) = member.ty else {
-        return;
-    };
-
     let Some(definition) = definition else {
         return;
     };
 
-    // Constructor methods are not checked for Liskov compliance
-    if matches!(
-        &*member.name,
-        "__init__" | "__new__" | "__post_init__" | "__init_subclass__"
-    ) {
-        return;
-    }
-
-    let (literal, specialization) = class.class_literal(db);
-    let class_kind = CodeGeneratorKind::from_class(db, literal, specialization);
-
-    // Synthesized `__replace__` methods on dataclasses are not checked
-    if &member.name == "__replace__"
-        && matches!(class_kind, Some(CodeGeneratorKind::DataclassLike(_)))
-    {
-        return;
-    }
-
     let Place::Defined(type_on_subclass_instance, _, _) =
         Type::instance(db, class).member(db, &member.name).place
     else {
         return;
     };
 
+    let (literal, specialization) = class.class_literal(db);
```

Comment:
Given that we do a ton of work here to check liskov, would it make sense to skip the work if the relevant rule is disabled?